website/server/models/group.js

```diff
@@ -724,6 +724,10 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     pusher.trigger(`presence-group-${group._id}`, 'user-left', {
       userId: user._id,
     });
+
+    delete group.quest.members[user._id];
+    group.markModified('quest.members');
+    promises.push(group.save());
```

Comment:
Best thing will probably to instantiate the update variable at the top of the function:

```js
let update = {
  $inc: {memberCount: -1},
};
```

And then here, $unset the user id from the members object.

```js
update.$unset = {[`quest.members.${user._id}`]: 1};
```

Then at the bottom, if the group is just removed, the update variable will get garbage collected and discarded. If it is updated, it'll include the call to unset the user from the quest.